### PR TITLE
Add Authoritative Server Only Replication

### PR DESCRIPTION
**Pull request by nurnbeck**
Description
Remerge of - https://github.com/spatialos/UnrealGDK/pull/3331
Added an auth server only replication type and updated labeled
applicable areas of the GDK.

Engine 4.26 change: improbableio/UnrealEngine#1061
Engine 4.27 change: improbableio/UnrealEngine#1058

Tests
How did you test these changes prior to submitting this pull request? In customer's codebase.
What automated tests are included in this PR?
updated SpatialGDKSchemaGeneratorTest

#### Documentation
release notes in readme

### CODE REVIEW

**Review by m-samiec — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestReplicationConditions/SpatialTestReplicationConditions.cpp [hunk 1]
```diff
@@ -506,6 +511,7 @@ void ASpatialTestReplicationConditions::ProcessCommonActorProperties(bool bWrite
 				  DynamicCompText);
 	WrappedAction(TestActor_Common->DynamicComponent->CondSkipReplay_Var, 540, COND_SkipReplay, DynamicCompText);
 	WrappedAction(TestActor_Common->DynamicComponent->CondServerOnly_Var, 550, COND_ServerOnly, DynamicCompText);
+	WrappedAction(TestActor_Common->DynamicComponent->CondAuthServerOnly_Var, 550, COND_AuthServerOnly, DynamicCompText);
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestReplicationConditions/SpatialTestReplicationConditions.cpp:514]:
```suggestion
	WrappedAction(TestActor_Common->DynamicComponent->CondAuthServerOnly_Var, 560, COND_AuthServerOnly, DynamicCompText);
```
Ultra nit

Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestReplicationConditions/SpatialTestReplicationConditions.cpp:514]:
```suggestion
	WrappedAction(TestActor_Common->DynamicComponent->CondAuthServerOnly_Var, 560, COND_AuthServerOnly, DynamicCompText);
```
Ultra nit

**Review by mironec — APPROVED** [review]
Seems good, we should also maybe create a ticket to convert some existing ServerOnly replication conditions in the engine into AuthServerOnly ones in an effort to save cpu/bandwidth.

**Comment by nurnbeck** [comment]
Put reverts for the engine PRs here if the build agent can't be fixed from the "done" state. either merging this or reverting those will resolve a failed build @m-samiec  https://github.com/improbableio/UnrealEngine/pull/1066 https://github.com/improbableio/UnrealEngine/pull/1065

**Pull request merged**